src/udp.rs

```diff
@@ -82,6 +82,13 @@ impl UdpSocket {
 
     /// Receives data from the socket. On success, returns the number of bytes
     /// read and the address from whence the data came.
+    ///
+    /// The function must be called with valid byte array `buf` of sufficient size to
```

Comment:
I think we still need to mention that this function only writes into the buffer and never reads from it.